commands.go

```diff
@@ -526,46 +526,55 @@ func (b *bot) ban(m dggchat.Message, s *dggchat.Session) {
 	}
 }
 
-// !roll sides [count] - roll dice
-func (b *bot) roll(m dggchat.Message, s *dggchat.Session) {
-	if !strings.HasPrefix(m.Message, "!roll") {
-		return
-	}
+func compute(input string) (int, error) {
 
-	parts := strings.Split(m.Message, " ")
-	if len(parts) < 2 {
-		return
+	// Define a regular expression to extract dice rolling information
+	regexPattern := `^!roll\s+(\d+)d(\d+)\s*([+\-]\s*\d+)?(.*?)$`
+	regex := regexp.MustCompile(regexPattern)
+
+	// Match the regular expression against the input string
+	matches := regex.FindStringSubmatch(input)
+
+	if matches == nil {
+		return 0, fmt.Errorf("Invalid input format: %s", input)
 	}
 
-	args := parts[1:]
+	// Extract matched values
+	numDice, _ := strconv.Atoi(matches[1])
+	numSides, _ := strconv.Atoi(matches[2])
+	modifierStr := matches[3]
 
-	// parse XdY
-	if strings.Contains(parts[1], "d") {
-		parts := strings.Split(parts[1], "d")
-		args = []string{parts[1], parts[0]}
+	if math.MaxInt64/numDice <= numSides || numDice > 100 {
```

Comment:
`math.MaxInt64/numDice <= numSides` is meant to check whether or not this could cause integer overflow. we also need to factor in the modifier now